dtrackScore panics with "index out of range" in v1.0.8 — works fine in v1.0.7

When running dtrackScore using version v1.0.8, the tool panics with an index out of range error. The same command works without issue in version v1.0.7.

Command:
`/app/sbomqs dtrackScore --url "https://${DEPENDENCY_TRACK_URL}/" --api-key "${DEPENDENCY_TRACK_API_KEY}" ${DEPENDENCY_TRACK_PROJECT_ID} --tag-project-with-score`

Error:
```
panic: runtime error: index out of range [0] with length 0
goroutine 1 [running]:
github.com/interlynk-io/sbomqs/pkg/engine.processFile({0xbd6e98, 0xc00011ae70}, 0xc0001caa38, {0xc000034e00, 0x39}, {0x0, 0x0}, {{0x0, 0x0}, {0x0, ...}, ...})
	/app/pkg/engine/score.go:316 +0x10f3
github.com/interlynk-io/sbomqs/pkg/engine.DtrackScore({0xbd6e98, 0xc00011ae70}, 0xc0000dc280)
	/app/pkg/engine/dtrack.go:92 +0x87b
github.com/interlynk-io/sbomqs/cmd.init.func3(0xf86660, {0xc0004341e0, 0x1, 0x6})
	/app/cmd/dtrackScore.go:48 +0x125
github.com/spf13/cobra.(*Command).execute(0xf86660, {0xc000434180, 0x6, 0x6})
	/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0xa94
github.com/spf13/cobra.(*Command).ExecuteC(0xf86ea0)
	/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x40c
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1064
github.com/charmbracelet/fang.Execute({0xbd6e60, 0xfb2ba0}, 0xf86ea0, {0x0, 0x0, 0x0?})
	/go/pkg/mod/github.com/charmbracelet/fang@v0.1.0/fang.go:131 +0x535
github.com/interlynk-io/sbomqs/cmd.Execute()
	/app/cmd/root.go:38 +0x2f
main.main()
	/app/main.go:20 +0xf
```

Workaround:
Downgrading to v1.0.7 resolves the issue.

Environment:
* [Container sbomqs version: v1.0.8](https://github.com/interlynk-io/sbomqs/pkgs/container/sbomqs/447275735?tag=v1.0.8) 
* Dependency-Track version: v4.13.2

--- a/pkg/engine/score.go
+++ b/pkg/engine/score.go
@@ -324,34 +324,35 @@ func processFile(ctx context.Context, ep *Params, path string, fs billy.Filesyst
 
 	sr := scorer.NewScorer(ctx, doc)
 
-	if len(ep.Features) > 0 && len(ep.Categories) > 0 {
-		fmt.Println("Using both features and categories, scoring will be done with mix of both")
-		for _, cat := range ep.Categories {
-			if len(cat) <= 0 {
-				continue
-			}
-			for _, feat := range ep.Features {
-				if len(feat) <= 0 {
+	if len(ep.Categories) > 0 {
+		if len(ep.Features) == 0 {
+			for _, category := range ep.Categories {
+				if len(category) <= 0 {
 					continue
 				}
 				filter := scorer.Filter{
-					Name:     feat,
-					Ftype:    scorer.Mix,
-					Category: cat,
+					Name:  category,
+					Ftype: scorer.Category,
 				}
 				sr.AddFilter(filter)
 			}
-		}
-	} else if len(ep.Categories) > 0 {
-		for _, category := range ep.Categories {
-			if len(category) <= 0 {
-				continue
+		} else if len(ep.Features) > 0 {
+			for _, cat := range ep.Categories {
+				if len(cat) <= 0 {
+					continue
+				}
+				for _, feat := range ep.Features {
+					if len(feat) <= 0 {
+						continue
+					}
+					filter := scorer.Filter{
+						Name:     feat,
+						Ftype:    scorer.Mix,
+						Category: cat,
+					}
+					sr.AddFilter(filter)
+				}
 			}
-			filter := scorer.Filter{
-				Name:  category,
-				Ftype: scorer.Category,
-			}
-			sr.AddFilter(filter)
 		}
 	} else if len(ep.Features) > 0 {
 		for _, feature := range ep.Features {

